src/plugins/visualizations/kibana.json

```diff
@@ -3,6 +3,6 @@
   "version": "kibana",
   "server": true,
   "ui": true,
-  "requiredPlugins": ["data", "expressions", "uiActions", "embeddable", "usageCollection", "inspector"],
+  "requiredPlugins": ["data", "expressions", "uiActions", "embeddable", "usageCollection", "inspector", "dashboard"],
```

Comment:
Are we certain the only way to address this is to introduce a required dependency between `visualizations` and `dashboard`?

This doesn't feel right to me, but I'm also not familiar enough with the `AttributeService` to understand whether alternatives are possible.

I just wanted to raise the question before we move forward with merging this PR.

Reply:
Hey Luke! This is a really good point.

I made the decision to move `attributeService` out of the embeddable plugin and temporarily into the dashboard plugin in https://github.com/elastic/kibana/pull/74302. It was moved because it makes use of the `savedObjectsClient`, and @streamich and I had a discussion about how best to avoid coupling `savedObjects` and `embeddables`.

The attributeService is an additional layer of abstraction on top of savedObjects. It provides a default implementation that embeddables can use to deal more easily with input that can be either 'by value' or 'by reference' . It also provides code for translating between the two types of input. 

The dashboard plugin is not its permanent home, however, it needs to stay somewhere until we find a better place for it.

Do you have any thoughts on where it could live?

Reply:
hmm, right i didn't notice this ...
saved objects is its own plugin, why did we prefer having attribute service in dashboard plugin instead of embeddable ? it seems embeddable specific ? if we don't want to add saved objects dependency to embeddable maybe this should be its own plugin ? shouldn't be a problem now that we can group plugins under domain folders

Reply:
Attribute service is indeed embeddable specific, but it's also only used within the dashboard (at least for now). I don't think saved objects plugin is the best place for it, since it's more than just saved objects. If there are major concerns about this living inside the dashboard, I think a separate plugin is the best way to go.
I am going to merge this PR now as much of the logic will remain unaffected regardless of where the attribute service lives. If concerns exists, I will revisit this in a follow-up PR.